### ISSUE TITLE
CropCircleTransformation can't work on glide4.0

### ISSUE THREAD

**Issue by 2391645784** [wasabeef/glide-transformations#82]
when i use the glide 4.0 ,it can't show the circle imagview

**Comment by lizihanglove** [comment]
you can just call the method apply  and you will get all things done like this:
`      RequestOptions requestOptions = new RequestOptions();
        CropCircleTransformation transformation = new CropCircleTransformation(HomeActivity.this);
        requestOptions.bitmapTransform(transformation);
        Glide.with(this).load("url").apply(requestOptions);`

**Comment by 2391645784** [comment]
@lizihanglove  I try do like this ,but it seem not avaliable
the code:
    RequestOptions requestOptions = new RequestOptions();
        requestOptions.bitmapTransform( new CropCircleTransformation(context));
        Glide.with(context).load(imgUrl).apply(requestOptions).into(imageView);

**Comment by 494293346** [comment]
I've tried it also. If there's an answer, please let me know

**Comment by gaurav-m** [comment]
@2391645784 @494293346 
The new glide 4.0 transform implementation is different than before. These glide-transforms need to get updated.

Create your own class which extends the CropCircleTransformation and override new methods

```
public class MyCropCircleTransformation extends CropCircleTransformation {
@Override
    public int hashCode() {
        return getId().hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        return (obj instanceof MyCropCircleTransformation); 
    }
    @Override
    public void updateDiskCacheKey(MessageDigest messageDigest) {
        try {
            byte[] ID_BYTES = getId().getBytes(STRING_CHARSET_NAME);
            messageDigest.update(ID_BYTES);
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
        }
    }
}

```
This will get the job done.
Alternatively if you are only looking for circle crop, I'll suggest you use Glide's own CircleCrop Transformation. No need to use external glide-transformations

**Comment by wasabeef** [comment]
I just released glide-transformations 3.0.0. So could you check that?
